lib/roby/app.rb

```diff
@@ -3032,12 +3039,29 @@ def autodiscover_tests_in?(path)
             end
         end
 
-        # Enumerate all the test files in this app and for this robot
+        # Enumerate all the test files in a specific dir for this robot
         # configuration
-        def each_test_file_in_app
+        def each_test_file_in(dir)
+            return enum_for(__method__) unless block_given?
```

Comment:
`enum_for(__method__, dir)`

enum_for in this case is meant to return an enumerator that corresponds to the full `each_...` call so that you can do stuff on it right away, e.g. `each_test_file_in(base_dir).each_with_object ...`

Reply:
`each_test_file_in(base_dir).each_with_object({})` returns:

```console
= wrong number of arguments (given 0, expected 1) (ArgumentError)
= Backtrace
| 
|   /home/caioaamaral/workspaces/wetpaint/tools/roby/lib/roby/app.rb:3043:in `each_test_file_in'
|   /home/caioaamaral/workspaces/wetpaint/tools/roby/lib/roby/app.rb:3005:in `each_with_object'
|   /home/caioaamaral/workspaces/wetpaint/tools/roby/lib/roby/app.rb:3005:in `discover_test_files'
```

Reply:
i think i need a way of injecting "appending_args" just like `each` allows https://ruby-doc.org/core-2.7.1/Enumerator.html#method-i-each

Reply:
maybe `test_files = each_test_file_in(base_dir).inject({}) do |h, t|` ?